src/main/java/com/epam/aidial/cfg/web/controller/ModelController.java

```diff
@@ -35,39 +38,43 @@ public ModelController(ModelFacade modelFacade) {
 
 
     @GetMapping(produces = MimeTypeUtils.APPLICATION_JSON_VALUE)
-    public Collection<ModelDto> getAllModels(HttpServletResponse response) throws Exception { //TODO change to model info
+    public Collection<ModelDto> getAllModels(HttpServletResponse response) { //TODO change to model info
         Collection<ModelDto> allModels = modelFacade.getAll();
         return allModels;
     }
 
     @GetMapping(path = "/{modelName}",
             produces = MediaType.APPLICATION_JSON_VALUE)
-    public ModelDto getModel(HttpServletResponse response,
-                             @PathVariable("modelName") String modelName) throws Exception {
-        var result = modelFacade.getModel(modelName);
-        return result;
+    public ResponseEntity<ModelDto> getModel(HttpServletResponse response,
+                                             @PathVariable("modelName") String modelName,
+                                             @RequestHeader(value = "If-None-Match") String previousHash) {
```

Comment:
let's mark PR as breaking change and add description of it's changes (helps build release notes)
also  update docs/samle/http-requests/AdminPanel.http